crates/nu-cli/src/commands/date/to_timezone.rs

```diff
@@ -0,0 +1,115 @@
+use crate::commands::date::parser::{get_timezone_offset, ParseErrorKind};
+use crate::commands::WholeStreamCommand;
+use crate::prelude::*;
+use nu_errors::ShellError;
+use nu_protocol::{Primitive, ReturnSuccess, Signature, SyntaxShape, UntaggedValue, Value};
+use nu_source::Tagged;
+
+pub struct Date;
+
+#[derive(Deserialize)]
+struct DateToTimeZoneArgs {
+    timezone: Tagged<String>,
+}
+
+#[async_trait]
+impl WholeStreamCommand for Date {
+    fn name(&self) -> &str {
+        "date to-timezone"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("date to-timezone").required(
+            "time zone",
+            SyntaxShape::String,
+            "time zone description",
+        )
+    }
+
+    fn usage(&self) -> &str {
+        "Convert a date to a given time zone.
+        
+Supported time zone abbreviations include: GMT, UTC, UT, EDT, EST, CDT, CST, MDT, MST, PDT, PST, LOCAL.
```

Comment:
This is another one we might be able to use a crate or built-in date function to help, so that we can support all the timezones.

Reply:
I don't think it's possible to support all time zone abbreviations since the same abbreviation can mean different time zones (also discussed in [this issue](https://github.com/chronotope/chrono-tz/issues/44)). For example, `CST` could mean "Central Standard Time" and "China Standard Time". Therefore, I've only included obsolete time zones supported in [RFC 2822](https://tools.ietf.org/html/rfc2822) and added `UTC` and `LOCAL` because they were mentioned in the discussions. ISO 8601/RFC 3339 doesn't support abbreviations at all.

Reply:
@jz448 - I see two resolutions to this. Please let me know what you think.
1. Doing a bit more research to see if there's a better way to handle timezones that somehow disambiguates the abbreviations.
2. Making the timezone function handle things more verbosely like `America/Chicago`.
3. ?
Would you be willing to look into this further?

Reply:
@fdncred If we use disambiguous time zone descriptions like `America/Chicago` instead of abbreviations, then it would be trivial to use the `chrono-tz` crate to parse the time zone. This means users would need to use `date to-timezone Europe/London` instead of `date to-timezone GMT`. Although it's longer to type, it removes the ambiguity and the need to remember the time zone abbreviations. Should I go for this option?

If we still want to support time zone abbreviations, one solution I can think of is to provide users a list of ambiguous time zones' full name and ask users to choose from it, but I don't think `Nushell` currently supports such interactive session.

Reply:
@jz488 yes, i think we should move ahead with a more verbose time zone functionality in order to disambiguate and still keep this feature in the PR that you've worked so hard on. we want to be as inclusive as possible, globally. so, i think the best way to do this is to use something the `chrono-tz` already supports. i looked into that crate a bit and it looks like it supports every time zone on the planet. i'm just not sure how users figure out what time zone to use. maybe there's a functionality that we do something like list the time zones in north america, austrailia, new zealand, south america, etc so people know what timezone names are valid. if listing isn't an option then i guess we just rely on them to know what to use somehow. thoughts?

Reply:
@fdncred Sounds like a good idea! I had another look at the `chrono-tz` crate and it's definitely viable to do this. We can add another command `date list-timezones`, which prints a table of all the time zones supported. The table has two columns: `timezone` and `region`. The value of `region` is None if the timezone doesn't match to a region.  This way, if users want only time zones related to a particular region (e.g., `Europe`), they can run something like `date list-timezones | where region == Europe`. Thoughts?

Reply:
Excellent!  Make it so, number one. :)